keps/sig-ui/5008-kep-template/README.md

```diff
@@ -0,0 +1,949 @@
+in-cluster web user interface, both based on the [Headlamp](https://headlamp.dev)
+project.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+The driving motivation behind this proposal is to provide both a great out of
+the box experience for end users and a foundation for vendors and other
+projects to integrate with and build upon. Essentially, we think Headlamps'
+ability to be adapted and used across many scenarios allows it to serve as a
+central hub for the CNCF community and projects.
+
+### Background
+
+Headlamp is a Kubernetes UI that can run as a Desktop application and an
+in-cluster web interface. It is designed to be extended using its robust plugin
+system. This allows vendors or projects to adapt it to their use case and for
+the project to deliver builds for various use cases.
+
+Headlamp is currently available in two versions:
+* Headlamp refers to the project and the default distribution of Headlamp which
+  ships with core Kubernetes UI functionality (see the next bullet point) plus
+  a set of plugins that the project thinks are generally useful for all users
+  in order to provide a rich and relevant Kubernetes user experience.
+* Headlamp Base does not include any plugins. It only includes features that
+  the project deems universally desired in a Kubernetes UI. This is designed
+  for builders looking to have a solid base on which to extend the user
+  experience. They would generally create custom plugins to customize the user
+  experience, include any existing plugins they desire and then theme the UI to
+  fit their branding.
+
+Each version is available as a desktop experience and an in-cluster experience.
+For the default build, the set of plugins differs between the in-cluster and
+desktop installations as there are different user profiles for each environment.
+For example, the plugin that allows managing plugins is not included in the
+in-cluster environment as the cluster admin generally determines which plugins
+to provide to the cluster users.
+
+### Existing Dashboard project
+
+The Kubernetes project already has a Dashboard project. However, we think that
+the Headlamp project has a number of advantages that make it a great solution
+to build on going forward. Some of these are:
+* Has a robust plugin system for customization and integrations
+* Supports both desktop and in-cluster mode
+* Is under very active development and has dedicated resources assigned to it
+  in addition to the active external contributors
+* Is built on a very common technology stack (Typescript, React, Material UI,
+  Vite, Go)
```

Comment:
Kubernetes Dashboard uses the same stack with Angular replacing React being the only difference.